frontend_new/src/components/BookmarkDrawer.tsx

```diff
@@ -1,87 +1,199 @@
 import React from "react";
 import clsx from "clsx";
+import {
+  makeStyles,
+  useTheme,
+  Theme,
+  createStyles,
+} from "@material-ui/core/styles";
 import Drawer from "@material-ui/core/Drawer";
-import { makeStyles } from "@material-ui/core/styles";
+import CssBaseline from "@material-ui/core/CssBaseline";
+import AppBar from "@material-ui/core/AppBar";
+import Toolbar from "@material-ui/core/Toolbar";
+import List from "@material-ui/core/List";
+import Typography from "@material-ui/core/Typography";
+import Divider from "@material-ui/core/Divider";
 import IconButton from "@material-ui/core/IconButton";
-import SettingsOutlinedIcon from "@material-ui/icons/SettingsOutlined";
+import MenuIcon from "@material-ui/icons/Menu";
+import ChevronLeftIcon from "@material-ui/icons/ChevronLeft";
+import ChevronRightIcon from "@material-ui/icons/ChevronRight";
+import ListItem from "@material-ui/core/ListItem";
+import ListItemIcon from "@material-ui/core/ListItemIcon";
+import ListItemText from "@material-ui/core/ListItemText";
+import InboxIcon from "@material-ui/icons/MoveToInbox";
+import MailIcon from "@material-ui/icons/Mail";
 import BookmarksOutlinedIcon from "@material-ui/icons/BookmarksOutlined";
-import List from "@material-ui/core/List";
+import Switch from "@material-ui/core/Switch";
+import Search from "./Search";
+import ThemeSwitch from "./ThemeSwitch";
+import Grow from "@material-ui/core/Grow";
+import Grid from "@material-ui/core/Grid";
+import SimpleCard from "./Cards.tsx";
 
-const useStyles = makeStyles((theme) => ({
-  root: {
-    flexGrow: 1,
-  },
-  menuButton: {
-    marginRight: theme.spacing(2),
-  },
-  title: {
-    flexGrow: 1,
-  },
-  list: {
-    width: 300,
-  },
-  fullList: {
-    width: "auto",
-  },
-}));
+const drawerWidth = "100%";
 
-type Anchor = "bookmark";
+const useStyles = makeStyles((theme: Theme) =>
```

Comment:
**[Suggestion]** Move this to another file perhaps?

Reply:
why?

Reply:
This is a pet peeve of mine about react, everything is in a single file. It's nicer to have it split by template, code and styling. But in light of React's style, seems fair to keep in file. It is fairly large though

Reply:
Yea my reasoning was that this whole section of style is too long, but yeah we can leave it in the same file for now

Reply:
Please resolve then